src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Memcached/MemcachedHelpers.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using NewRelic.Agent.Api;
+using NewRelic.Agent.Extensions.Parsing;
+using NewRelic.Agent.Extensions.Providers.Wrapper;
+using NewRelic.Reflection;
+
+namespace NewRelic.Providers.Wrapper.Memcached
+{
+    public class MemcachedHelpers
+    {
+        private static bool _hasGetServerFailed = false;
+        private const string AssemblyName = "EnyimMemcachedCore";
+        private static Func<object, object> _transformerGetter;
+        private static Func<object, string, string> _transformMethod;
+        private static Func<object, object> _poolGetter;
+        private static Func<object, string, object> _locateMethod;
+        private static Func<object, object> _endpointGetter;
+        private static Func<object, object> _addressGetter;
+        private static Func<object, int> _portGetter;
+
+        // To get the ConnectionInfo we need to call the same Transform method that the library calls to get the node.
+        // This is deterministic based reviewing the code at different versions.
+        public static ConnectionInfo GetConnectionInfo(string key, object target, IAgent agent)
+        {
+            if (_hasGetServerFailed)
+            {
+                return new ConnectionInfo(DatastoreVendor.Memcached.ToKnownName(), null, -1, null);
+            }
+
+            try
+            {
+                var targetType = target.GetType();
```

Comment:
Is this type ever a generic type? Or is this always a single type for the entire application? The caching used here assumes that this is always a single type.

Reply:
Based on reading through the library code, it should be a single type for the duration of the app.